src/test/java/cat/udl/eps/softarch/tfgfinder/steps/CreateCategoryStepDefs.java

```diff
@@ -0,0 +1,65 @@
+package cat.udl.eps.softarch.tfgfinder.steps;
+
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import jakarta.validation.ConstraintViolationException;
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import cat.udl.eps.softarch.tfgfinder.domain.Category;
+import cat.udl.eps.softarch.tfgfinder.repository.CategoryRepository;
+
+import java.util.Optional;
+
+@SpringBootTest
+public class CreateCategoryStepDefs {
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    private Category category;
+    private Optional<Category> retrievedCategory;
+
+    @Given("a category with name {string} and description {string}")
+    public void a_category_with_name_and_description(String name, String description) {
+        category = new Category();
+        category.setName(name);
+        category.setDescription(description);
+    }
+
+    @When("I save the category")
```

Comment:
The When step should use the API to test if the backend is working as expected. Otherwise, we are testing the Spring Framework but not the application we are developing.